ra/ra.go

```diff
@@ -1926,6 +1942,7 @@ func (ra *RegistrationAuthorityImpl) purgeOCSPCache(ctx context.Context, cert *x
 	if err != nil {
 		return err
 	}
+	purgeURLs = uniquePurgeURLs(purgeURLs)
```

Comment:
This does correctly ensure that the set of URLs sent from the RA to the akamai-purger in a single request contains no duplicates.

But the problem isn't that we're sending duplicate URLs in a single request. The problem is that we're sending the same request _twice_.

Line [2126](https://github.com/letsencrypt/boulder/blob/d6192e7c56fbd69063183fee6cc6e50e1afe93cc/ra/ra.go#L2126) and line [2146](https://github.com/letsencrypt/boulder/blob/d6192e7c56fbd69063183fee6cc6e50e1afe93cc/ra/ra.go#L2146) both independently call `ra.purgeOCSPCache` (this function), so the contents of those two purge requests won't be de-duplicated against each other.

The fix here isn't going to be deduplicating the contents of the request, it's going to be refactoring `RevokeCertByKey` so that it doesn't call `ra.purgeOCSPCache` twice in a row.

Reply:
Ahhhhhhhhh 😵‍💫